cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -207,6 +208,160 @@ func Run(c *config.CompletedConfig, stopCh <-chan struct{}) error {
 		}
 	}
 
+	// If leader migration is enabled, use the redirected initialization
+	// otherwise, make absolutely NO change to the initialization process
+	if leadermigration.FeatureEnabled() && c.ComponentConfig.Generic.LeaderMigrationEnabled &&
```

Comment:
Should we inline this code here or break it out into a separate function? I see that the code does follow the conventions of the surrounding code (e.g. the nested function declarations) which is fine, but maybe a separate `runLeaderMigration` function would be a better place for this to all live?

Reply:
I think leave it like this shows clearly the condition where leader migration will be enabled, and thus it is better than extracted into a func.